core/src/test/java/org/frankframework/pipes/MoveFilePipeTest.java

```diff
@@ -1,366 +1,318 @@
-        pipe.setDeleteEmptyDirectory(true);
-        pipe.setFilename("deletable.sd");
-        pipe.configure();
-        pipe.start();
-
-        PipeRunResult res = doPipe(pipe, "xx", session);
-
-        assertEquals(pipeForwardThen, res.getPipeForward().getName());
-    }
-
-
-
-    @Test
-    @Order(19)
-    public void failCreatingNewDirectory() throws ConfigurationException, PipeStartException, PipeRunException {
-        exception.expect(PipeRunException.class);
-        pipe.setCreateDirectory(false);
-        pipe.setMove2dir(destFolderPath+"/newas");
-        pipe.setDirectory(sourceFolderPath);
-        pipe.setFilename("sad.lk");
-        pipe.setNumberOfAttempts(1);
-        pipe.configure();
-        pipe.start();
-
-        doPipe(pipe, "xx", session);
-
-        fail("this is expected to fail");
-    }
+	@TempDir
+	public static Path testFolderSource;
+
+	private static String sourceFolderPath;
+
+	@TempDir
+	public static Path testFolderDest;
+
+	private static String destFolderPath;
+
+	@TempDir
+	public static Path testFolderCantDelete;
+
+	private static String cantdeleteFolderPath;
+
+	@TempDir
+	public static Path testFolderDelete;
+
+	private static String deleteFolderPath;
+
+	@Override
+	public MoveFilePipe createPipe() {
+		return new MoveFilePipe();
+	}
+
+	@BeforeAll
+	public static void setUpTest() throws IOException {
+		sourceFolderPath = testFolderSource.toString();
+		destFolderPath = testFolderDest.toString();
+		cantdeleteFolderPath = testFolderCantDelete.toString();
+		deleteFolderPath = testFolderDelete.toString();
+
+		Files.createFile(testFolderSource.resolve("1.txt"));
+		Files.createFile(testFolderSource.resolve("2.txt"));
+		Files.createFile(testFolderSource.resolve("3.txt"));
+		Files.createFile(testFolderSource.resolve("a.md"));
+		Files.createFile(testFolderSource.resolve("b.md"));
+		Files.createFile(testFolderSource.resolve("sad.lk"));
+		Files.createFile(testFolderSource.resolve("notCompatible.txt"));
+		Files.createFile(testFolderSource.resolve("createDirectory.txt"));
+		Files.createFile(testFolderSource.resolve("cantmove.sc"));
+		Files.createFile(testFolderSource.resolve("prefixsuffix.txt"));
+		Files.createFile(testFolderSource.resolve("toAppend1.txt"));
+		Files.createFile(testFolderSource.resolve("toAppend2.txt"));
+		Files.createFile(testFolderSource.resolve("xx.txt"));
+		Files.createFile(testFolderSource.resolve("test.txt"));
+		Files.createFile(testFolderDest.resolve("cantmove.sc"));
+		Files.createFile(testFolderDest.resolve("notcompatible.asd"));
+		Files.createFile(testFolderDest.resolve("toBeAppended.txt"));
+		Files.createFile(testFolderCantDelete.resolve("deletable.sd"));
+		Files.createFile(testFolderCantDelete.resolve("deletionInterrupter.mz"));
+		Files.createFile(testFolderDelete.resolve("moveAndDeleteDirectory.txt"));
+	}
+
+	@Test
+	public void nonExistingFileWithSourceAndTargetDirectories() throws ConfigurationException, PipeStartException, PipeRunException {
```

Comment:
Since you've removed the `@Order` annotations, does this class now run fine in any random order? The directory setup is only performed once, and the tests don't alter the content?

Reply:
It's an annotation from `Spring`, it has nothing to do with tests, and has never worked.

Reply:
Ah.. I assumed it's from JUnit 😉 Thanks.